### PR TITLE
Add associations

### PR DESCRIPTION
**Pull request by SFarida**
### This PR contains the following
- a class `Classroom` with an instance variable `@label` and a setter and getter for `@label`
- the `has-many/belongs-to` relationship between `Classroom` and `Student`
- a class `Book` with `@title` and `@author` (with setters and getters) as instance variables initialized in the constructor
- a class `Rental` with `@date` (with a setter and getter) instance variable initialized in the constructor.
- the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Rental
+  attr_accessor :date
+  attr_reader :person, :book
```
Comment (prolajumokeoni) [inline rental.rb]: - [ ] Kindly add your person and book variables to attribute accessor, based on the requirements, `Getters and Setters for @book and @person instance variables should be present`

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :author, :title
+  attr_reader :rental
```
Comment (prolajumokeoni) [inline book.rb]: - [ ] Kindly add your rental variable to attribute accessor,  based on the requirements `Getters and Setters for @rentals should be present`

File: person.rb [hunk 3]
```diff
@@ -1,16 +1,17 @@
 require './nameable'
 class Person < Nameable
+  attr_reader :rental, :id
```
Comment (prolajumokeoni) [inline person.rb]: - [ ] Kindly add your rental variable to attribute accessor,  based on the requirements `Getters and Setters for @rentals should be present`

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @SFarida 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Associations set up 



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 3 above)
Comment (SFarida) [reply person.rb]: Hello @prolajumokeoni thank you for your remarks I'll correct them right away

File: rental.rb (hunk 1 above)
Comment (KDlamini) [reply rental.rb]: Well resolved ✔️

File: book.rb (hunk 2 above)
Comment (KDlamini) [reply book.rb]: Well resolved ✔️

File: person.rb (hunk 3 above)
Comment (KDlamini) [reply person.rb]: Well resolved ✔️

**Review by KDlamini — APPROVED** [review]
### Hi @SFarida 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## HIGHLIGHTS 🌟 

✔️ Good PR title and description
✔️ Relationships and associations between classes are implemented
✔️ Getters and setters are created using attr_accessor
✔️ All required methods are well implemented

# Status: Approved ✔️ 

oscars-standing-ovation
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**